File: bumperbot_py_examples/bumperbot_py_examples/simple_parameter.py
import rclpy 
from rcl_interfaces.msg import SetParametersResult
from rclpy.parameter import Parameter
from rclpy.node import Node 


class SimpleParameter(Node): 
    def __init__(self):
        super().__init__("simple_parameter")

        self.get_logger().info("O nó 'SimpleParameter' foi iniciado com sucesso!")
        self.declare_parameter("simple_int_parameter", 28) 
        self.declare_parameter("simple_string_parameter", "Matheus")
        self.add_on_set_parameters_callback(self.paramChangeCallback)


    def paramChangeCallback(self, params):
        result = SetParametersResult()
        for param in params: 
            if param.name == "simple_int_parameter" and param.type_ == Parameter.Type.INTEGER: 
                self.get_logger().info("O parametro 'simple_int_param' foi alterado!, e o novo valor é: %d", param.value)
                result.successful = True

            if param.name == "simple_string_parameter" and param.type_ == Parameter.Type.STRING: 
                self.get_logger().info("O parametro 'simple_string_param' foi alterado!, e o novo valor é: %s", param.value)
                result.successful = True
        
            
            
    
        return result 
    


def main():
    rclpy.init()
    simple_parameter = SimpleParameter()
    rclpy.spin(simple_parameter)
    simple_parameter.destroy_node()
    rclpy.shutdown()

if __name__ == "__main__":
    main()

File: bumperbot_cpp_examples/src/simple_parameter.cpp
#include <rclcpp/rclcpp.hpp> 
#include <string>
#include <vector>
#include <rcl_interfaces/msg/set_parameters_result.hpp>
#include <memory>
using namespace std; 
using namespace placeholders;

class SimpleParameter : public rclcpp::Node
{

public:
    SimpleParameter() : Node("simple_parameter")
    {   
        RCLCPP_INFO(this->get_logger(),"O nó 'SimpleParameter' foi iniciado com sucesso!");
        declare_parameter<int>("simple_int_param", 28); 
        declare_parameter<string>("simple_string_param", "Matheus");
        param_callback_handle_ = add_on_set_parameters_callback(bind
            (&SimpleParameter::paramChangeCallback, this, _1));


    }

private: 
    OnSetParametersCallbackHandle:: SharedPtr param_callback_handle_;

   rcl_interfaces ::msg::SetParametersResult paramChangeCallback(const vector<rclcpp::Parameter> & parameters)
    {
        rcl_interfaces::msg::SetParametersResult result;
        for(const auto& param: parameters)
        {
            if(param.get_name() == "simple_int_param" && param.get_type() == rclcpp::ParameterType::PARAMETER_INTEGER)
            {
                RCLCPP_INFO_STREAM(get_logger(),"Parametro 'int' foi trocado. Novo valor = " <<param.as_int());
                result.successful = true; 
            }

              if(param.get_name() == "simple_string_param" && param.get_type() == rclcpp::ParameterType::PARAMETER_STRING)
            {
                RCLCPP_INFO_STREAM(get_logger(),"Parametro 'string' foi trocado. Novo valor = " <<param.as_string());
                result.successful = true; 
            }


        }

        return result;

    }
};


int main (int argc, char* argv[])
{
    rclcpp::init(argc,argv);
    auto node = make_shared<SimpleParameter>();
    rclcpp::spin(node);
    rclcpp::shutdown();
    return 0;
}
